### PR TITLE
Fix MaxRevalDurationDays validation for invalidation jobs

### PR DESCRIPTION
**Pull request by rawlinp**
## What does this PR (Pull Request) do?

Fix the validation of invalidation job TTLs per the `maxRevalDurationsDays` parameter. The broken validation was not converting days to hours before doing the validation. This PR also adds that validation to the `/jobs` endpoints because it was missing.

- [x] This PR is not related to any Issue

## Which Traffic Control components are affected by this PR?

- Traffic Ops

## What is the best way to verify this PR?

Run the unit tests and TO API tests. Verify they pass. Manually `POST /api/1.4/user/current/jobs` with a ttl that higher than the value of the `maxRevalDurationDays` parameter (x 24). Example request json below:
```
{
  "startTime": "2020-05-26 16:48:32",
  "dsId": 442,
  "regex": "/foobar",
  "ttl": 3000
}
```

## If this is a bug fix, what versions of Traffic Control are affected?
- master
- 4.0.0
- 4.1.0-RC0

## The following criteria are ALL met by this PR

- [x] This PR includes tests
- [x] bugfix, no docs necessary
- [x] This PR includes an update to CHANGELOG.md
- [x] This PR includes any and all required license headers
- [x] This PR ensures that database migration sequence is correct OR this PR does not include a database migration
- [x] This PR **DOES NOT FIX A SERIOUS SECURITY VULNERABILITY** (see [the Apache Software Foundation's security guidelines](https://www.apache.org/security/) for details)

### CODE REVIEW

**Comment by zrhoffman** [comment]
Branch has a merge conflict in the changelog

**Comment by rawlinp** [comment]
Thanks! Merge conflict resolved.

File: traffic_ops/testing/api/v3/jobs_test.go [hunk 1]
```diff
@@ -82,6 +83,28 @@ func CreateTestInvalidationJobs(t *testing.T) {
...
+	job := testData.InvalidationJobs[0]
+	_, ok := dsNameIDs[(*job.DeliveryService).(string)]
+	if !ok {
+		t.Fatalf("can't create test data job: delivery service '%v' not found in Traffic Ops", job.DeliveryService)
+	}
+	tooHigh := interface{}(2161)
+	job.TTL = &tooHigh
+	if _, _, err := TOSession.CreateInvalidationJob(job); err == nil {
+		t.Error("creating invalid job (TTL higher than maxRevalDurationDays) - expected: error, actual: nil error")
+	}
```
Comment (zrhoffman) [inline traffic_ops/testing/api/v3/jobs_test.go:105]: Should assert the status code is 400-level

File: lib/go-tc/invalidationjobs.go [hunk 2]
```diff
@@ -349,12 +356,13 @@ func (job *UserInvalidationJobInput) Validate(tx *sql.Tx) error {
...
-		err := row.Scan(&max)
+		var maxDays uint64
+		err := row.Scan(&maxDays)
+		maxHours := maxDays * 24
 		if err == sql.ErrNoRows && MaxTTL < *(job.TTL) {
 			errs = append(errs, "ttl: cannot exceed "+strconv.FormatUint(MaxTTL, 10)+"!")
-		} else if err == nil && max < *(job.TTL) { //silently ignore other errors to
-			errs = append(errs, "ttl: cannot exceed "+strconv.FormatUint(max, 10)+"!")
+		} else if err == nil && maxHours < *(job.TTL) { //silently ignore other errors to
+			errs = append(errs, "ttl: cannot exceed "+strconv.FormatUint(maxHours, 10)+"!")
```
Comment (zrhoffman) [inline lib/go-tc/invalidationjobs.go:365]: Errors should not end with punctuation

File: lib/go-tc/invalidationjobs.go [hunk 3]
```diff
@@ -349,12 +356,13 @@ func (job *UserInvalidationJobInput) Validate(tx *sql.Tx) error {
...
-		var max uint64
-		err := row.Scan(&max)
+		var maxDays uint64
+		err := row.Scan(&maxDays)
+		maxHours := maxDays * 24
 		if err == sql.ErrNoRows && MaxTTL < *(job.TTL) {
 			errs = append(errs, "ttl: cannot exceed "+strconv.FormatUint(MaxTTL, 10)+"!")
-		} else if err == nil && max < *(job.TTL) { //silently ignore other errors to
-			errs = append(errs, "ttl: cannot exceed "+strconv.FormatUint(max, 10)+"!")
+		} else if err == nil && maxHours < *(job.TTL) { //silently ignore other errors to
```
Comment (zrhoffman) [inline lib/go-tc/invalidationjobs.go:364]: There's no better time to add an "o".

File: traffic_ops/testing/api/v3/jobs_test.go [hunk 4]
```diff
@@ -82,6 +83,28 @@ func CreateTestInvalidationJobs(t *testing.T) {
...
+	for _, ds := range toDSes {
+		dsNameIDs[*ds.XMLID] = int64(*ds.ID)
+	}
+
+	job := testData.InvalidationJobs[0]
+	_, ok := dsNameIDs[(*job.DeliveryService).(string)]
+	if !ok {
+		t.Fatalf("can't create test data job: delivery service '%v' not found in Traffic Ops", job.DeliveryService)
+	}
+	tooHigh := interface{}(2161)
```
Comment (zrhoffman) [inline traffic_ops/testing/api/v3/jobs_test.go:101]: 2160 is a well-known TTL, but this magic number should still be factored into `x * y * [...]`

File: traffic_ops/testing/api/v3/jobs_test.go [hunk 5]
```diff
@@ -82,6 +83,28 @@ func CreateTestInvalidationJobs(t *testing.T) {
 	}
 }
 
+func CreateInvalidJob(t *testing.T) {
```
Comment (zrhoffman) [inline traffic_ops/testing/api/v3/jobs_test.go:86]: To follow naming conventions, this should be `CreateTestInvalidJob()`.

File: lib/go-tc/invalidationjobs.go (hunk 3 above)
Comment (ocket8888) [reply lib/go-tc/invalidationjobs.go:364]: or an "h"? I don't remember

File: traffic_ops/testing/api/v3/jobs_test.go (hunk 1 above)
Comment (rawlinp) [reply traffic_ops/testing/api/v3/jobs_test.go]: done in bae7ca548739e0d9fe501afba8b2c2559b7843f9

File: lib/go-tc/invalidationjobs.go (hunk 2 above)
Comment (rawlinp) [reply lib/go-tc/invalidationjobs.go]: done in bae7ca548739e0d9fe501afba8b2c2559b7843f9

File: lib/go-tc/invalidationjobs.go (hunk 3 above)
Comment (rawlinp) [reply lib/go-tc/invalidationjobs.go]: done in bae7ca548739e0d9fe501afba8b2c2559b7843f9

File: traffic_ops/testing/api/v3/jobs_test.go (hunk 4 above)
Comment (rawlinp) [reply traffic_ops/testing/api/v3/jobs_test.go]: done in bae7ca548739e0d9fe501afba8b2c2559b7843f9

File: traffic_ops/testing/api/v3/jobs_test.go (hunk 5 above)
Comment (rawlinp) [reply traffic_ops/testing/api/v3/jobs_test.go]: done in bae7ca548739e0d9fe501afba8b2c2559b7843f9

File: traffic_ops/testing/api/v3/jobs_test.go [hunk 6]
```diff
@@ -82,6 +85,47 @@ func CreateTestInvalidationJobs(t *testing.T) {
...
+	maxRevalDays := 0
+	foundMaxRevalDays := false
+	for _, p := range testData.Parameters {
+		if p.Name == "maxRevalDurationDays" {
+			maxRevalDays, err = strconv.Atoi(p.Value)
+			if err != nil {
+				t.Fatalf("unable to parse maxRevalDurationDays value '%s' to int", p.Value)
+			}
+			foundMaxRevalDays = true
+			break
```
Comment (zrhoffman) [inline traffic_ops/testing/api/v3/jobs_test.go:112]: This block can be unindented if it's moved to after the `if` statement, the condition is negated, and the loop `continue`s if the condition is met

Comment (rawlinp) [reply traffic_ops/testing/api/v3/jobs_test.go]: unindented in 7509a9faeb20027557d9cc2332755dd47c8e8498

**Pull request merged**